subsys/usb/usb_device.c

```diff
@@ -549,9 +553,9 @@ static bool reset_endpoint(const struct usb_ep_descriptor *ep_desc)
 
 	usb_cancel_transfer(ep_cfg.ep_addr);
 
-	if (usb_dc_ep_disable(ep_cfg.ep_addr) < 0) {
-		LOG_ERR("Failed to disable endpoint 0x%02x", ep_cfg.ep_addr);
-		return false;
+	ret = usb_dc_ep_disable(ep_cfg.ep_addr);
+	if (ret && ret != -EALREADY) {
+		LOG_WRN("Failed to disable endpoint 0x%02x", ep_cfg.ep_addr);
```

Comment:
Should we return `false` if ret equal to other than `EALREADY ` error?

Reply:
Yes and probably keep `LOG_ERR("Failed...);` but log `ret != -EALREADY` brunch with `LOG_WRN("Endpoint 0x%02x already disabled", ep_cfg.ep_addr);`

Reply:
I think in this case we should also return false on similar conditions for ```usb_dc_ep_configure``` and ```usb_dc_ep_enable```

I will push a patch for that.